traveltimepy/requests/common.py

```diff
@@ -0,0 +1,376 @@
+    Controls whether the time/distance required to reach the actual road network is included in metrics.
+
+    Attributes:
+        ENABLED: Walking time and distance from departure to nearest road and from nearest road
+                to arrival are added to total travel time and distance. Provides realistic door-to-door metrics.
+        DISABLED: Walking times and distances are not added to reported values. Journey effectively
+                 starts and ends at nearest points on the road network.
+    """
+
+    ENABLED = "enabled"
+    DISABLED = "disabled"
+
+
+class SnappingAcceptRoads(str, Enum):
+    """
+    Defines which road types are valid for journey start/end points based on their accessibility characteristics.
+    This affects where a journey can begin or terminate within the routing network.
+
+    Attributes:
+        BOTH_DRIVABLE_AND_WALKABLE: Journey can only start or end on roads accessible by both cars and pedestrians.
+                                   Effectively means journeys cannot start/end on motorways or vehicle-only roads.
+        ANY_DRIVABLE: Journey can start or end on any road accessible by a car (including motorways).
+                     Maximizes vehicle routing options but may result in pedestrian-inaccessible endpoints.
+    """
+
+    BOTH_DRIVABLE_AND_WALKABLE = "both_drivable_and_walkable"
+    ANY_DRIVABLE = "any_drivable"
+
+
+class DrivingTrafficModel(str, Enum):
+    OPTIMISTIC = "optimistic"
+    BALANCED = "balanced"
+    PESSIMISTIC = "pessimistic"
+
+
+class ArrivalTimePeriod(str, Enum):
+    WEEKDAY_MORNING = "weekday_morning"
+
+
+class Snapping(BaseModel):
+    """
+    Configuration for how journey calculations handle connections between arbitrary points and the road network.
+
+    "Snapping" refers to the process of connecting departure/arrival locations to the nearest suitable roads
+    and determining how the off-road portions of journeys are calculated and reported.
+
+    Attributes:
+        penalty: Controls whether off-road walking distances are included in total journey metrics.
+                When enabled, includes "first and last mile" walking times/distances for realistic door-to-door calculations.
+        accept_roads: Defines which road types are valid as journey start/end points.
+                     Determines whether journeys can snap to vehicle-only roads or only pedestrian-accessible roads.
+    """
+
+    penalty: Optional[SnappingPenalty] = SnappingPenalty.ENABLED
+    accept_roads: Optional[SnappingAcceptRoads] = (
+        SnappingAcceptRoads.BOTH_DRIVABLE_AND_WALKABLE
+    )
+
+
+class ProtoProperty(int, Enum):
+    DISTANCE = 1
+
+
+class FullRange(BaseModel):
+    """
+    Configures time range search parameters for journey planning.
+
+    When enabled, allows searches to consider journeys within a specified
+    time window rather than at an exact time only, providing multiple journey options.
+    """
+
+    enabled: bool
```

Comment:
I think the documentation strings should be *before* the parameters, not after.

Reply:
If I move documentation string, then IDE doesn't find it. It would be possible to rework this to `docustring` though, but then probably documentation should be shorter.